src/faebryk/libs/picker/jlcpcb/picker_lib.py

```diff
@@ -497,6 +497,81 @@ def find_ldo(cmp: Module):
     )
 
 
+def find_header(cmp: Module):
+    """
+    Find a header part in the JLCPCB database that matches the parameters of the
+    provided header
+    """
+
+    assert isinstance(cmp, F.Header)
+
+    def pin_structure_to_pin_count(vertical: bool) -> Callable[[str], F.Constant[int]]:
+        """
+        Database data looks like: 1x3P
+        """
+
+        def f(x: str) -> F.Constant[int]:
+            if "x" not in x:
+                raise ValueError(f"Invalid pin structure: {x}")
+            if vertical:
+                return F.Constant(int(x.strip("P").split("x")[0]))
+            else:
+                return F.Constant(int(x.strip("P").split("x")[1]))
+
+        return f
+
+    mapping = [
+        MappingParameterDB(
+            "pin_pitch",
+            [
+                "Row Spacing",
+                "Pitch",
+            ],
+        ),
+        MappingParameterDB(
+            "mating_pin_lenght",
+            ["Length of Mating Pin"],
+        ),
+        MappingParameterDB(
+            "conection_pin_lenght",
+            ["Length of End Connection Pin"],
+        ),
+        MappingParameterDB(
+            "spacer_height",
+            ["Insulation Height"],
+        ),
+        # MappingParameterDB(
```

Comment:
Needs some mapping function between `F.Header.Angle` enum names and `"Mounting Type"` str as they are not the same